### PR TITLE
[MO][OPS-11096] Added 'Do you want to sign in?' page

### PR DESCRIPTION
**Pull request by Lyrain**

### CODE REVIEW

**Review by barnesjake — CHANGES_REQUESTED** [review]
Just in case we forget to change this later, we shouldn't point it to test only (although I'm not sure what else it should be)

**Comment by barnesjake** [comment]
> Just in case we forget to change this later, we shouldn't point it to test only (although I'm not sure what else it should be)

Sorry this was supposed to be on line 36 on the do you want to sign in twirl template file (I'm using GitHub on mobile)

File: app/language/LangMessages.scala [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+package language
+
+object LangMessages {
+
+  val doYouWantToSignInH1: Message = Message(
+    english = "Do you want to sign in?"
+  )
+
+  val doYouWantToSignInHint: Message = Message(
+    english = "Sign in with your Government Gateway user ID. You'll have fewer details to enter this way."
```
Comment (AndyHWChung) [inline app/language/LangMessages.scala:26]: is this a straight quote here? We should be using curly quotes like the ones described [here](https://smartquotesforsmartpeople.com/) for UI content

File: app/forms/DoYouWantToSignInForm.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+package forms
+
+import language.{ErrorMessages, Language}
+import play.api.data.Forms.{mapping, optional, text}
+import play.api.data.Form
+
+final case class DoYouWantToSignInForm(
+    signIn: Option[String]
```
[the remark this answers is not in the document]
Comment (barnesjake) [reply app/forms/DoYouWantToSignInForm.scala:24]: https://github.com/hmrc/essttp-frontend/blob/main/app/models/forms/CanPayUpfrontForm.scala

Maybe something like this usage would be nice

File: app/language/ErrorMessages.scala [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+package language
+
+object ErrorMessages {
+  val `There is a problem`: Message = Message(
+    english = "There is a problem",
+    welsh   = "Mae problem wedi codi"
+  )
+
+  val `Select yes if you want to sign in to your tax account`: Message = Message(
```
Comment (AndyHWChung) [inline app/language/ErrorMessages.scala:25]: don't know if this structure is deliberate but if I want to see all messages that can appear on a page having a separate `ErrorMessages` file makes that a bit difficult. Perhaps sticking everything in `LangMessages` and grouping together by page would make it easier, e.g.:
```scala
object LangMessages {

 Common {
    val continue: Message = ...
    val `There is a problem`: Message =
  }
  
 DoYouWantToSignIn {
    val `Sign in with your Government Gateway user ID.`: Message =
    val `Yes, sign in`: Message = ...
    val `No, continue without signing in`: Message = ...
    val `Select yes if you want to sign in to your tax account`: Message = ...
  }

  OtherPageInTheFuture { ... }

}
```
What do you think?

**Review by barnesjake — APPROVED** [review]

**Review by AndyHWChung — APPROVED** [review]

**Pull request merged**